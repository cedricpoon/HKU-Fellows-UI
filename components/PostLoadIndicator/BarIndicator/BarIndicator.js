import React, { PureComponent } from 'react';
import { Platform } from 'react-native';
import { View } from 'native-base';
import { Placeholder as Placeholder_ } from 'react-native-loading-placeholder';
import PropTypes from 'prop-types';

import styles from '../Styles';
import { makeAnimatable } from 'hkufui/components/helper';
import { FADE_IN_DURATION } from 'hkufui/components/Constants';

const Wrapper = makeAnimatable(View);
const Placeholder = Platform.OS === 'ios' ? Placeholder_ : View;

class BarIndicator extends PureComponent {

  render() {
    const { delay, stalled, paragraph } = this.props;

    const context = (
      <View>
        <View
          style={paragraph ? styles.paragraphGroup : styles.placeholderGroup}
        >
          <Placeholder
            style={[paragraph ? styles.placeholderNoMarginTop : styles.placeholder, styles.placeholderShort]}
          />
          <Placeholder
            style={[styles.placeholder, styles.placeholderLong]}
          />
          <Placeholder
            style={[styles.placeholder, styles.placeholderLong]}
          />
          {paragraph && (
            <View>
              <Placeholder
                style={[styles.placeholder, styles.placeholderLong]}
              />
              <Placeholder
                style={[styles.placeholder, styles.placeholderLong]}
              />
            </View>
          )}
        </View>
        {!paragraph && (
          <View
            style={styles.hr}
          />
        )}
      </View>
    );

    // no animation
    if (stalled) {
      return (
        <View>
          {context}
        </View>
      );
    }

    return (
      <Wrapper
        animation='fadeIn'
        duration={FADE_IN_DURATION}
        delay={delay}
      >
        {context}
      </Wrapper>
    );
  }
}

BarIndicator.defaultProps = {
  delay: 0
}

BarIndicator.propTypes = {
  delay: PropTypes.number,
  stalled: PropTypes.bool,
  paragraph: PropTypes.bool
}

export default BarIndicator;
