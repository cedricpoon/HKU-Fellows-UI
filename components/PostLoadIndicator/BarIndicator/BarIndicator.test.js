import React from 'react';
import { shallow } from 'enzyme';

import BarIndicator from './BarIndicator';

describe('Testing BarIndicator Component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <BarIndicator />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with delayed animation', () => {
    const wrapper = shallow(
      <BarIndicator delayed={1} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without animation', () => {
    const wrapper = shallow(
      <BarIndicator stalled />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with paragraph mode', () => {
    const wrapper = shallow(
      <BarIndicator paragraph />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
