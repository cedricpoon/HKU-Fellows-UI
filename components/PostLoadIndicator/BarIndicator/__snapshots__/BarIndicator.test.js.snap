// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BarIndicator Component renders as expected 1`] = `
<withAnimatable(Styled(ViewNB))
  animation="fadeIn"
  delay={0}
  direction="normal"
  duration={250}
  iterationCount={1}
  iterationDelay={0}
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onTransitionBegin={[Function]}
  onTransitionEnd={[Function]}
  useNativeDriver={false}
>
  <Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "marginBottom": 11.5,
          "marginLeft": 35,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "30%",
            },
          ]
        }
      />
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      />
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      />
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "backgroundColor": "#eee",
          "height": 1,
          "marginLeft": 35,
          "marginTop": 5,
          "width": "100%",
        }
      }
    />
  </Styled(ViewNB)>
</withAnimatable(Styled(ViewNB))>
`;

exports[`Testing BarIndicator Component renders with delayed animation 1`] = `
<withAnimatable(Styled(ViewNB))
  animation="fadeIn"
  delay={0}
  direction="normal"
  duration={250}
  iterationCount={1}
  iterationDelay={0}
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onTransitionBegin={[Function]}
  onTransitionEnd={[Function]}
  useNativeDriver={false}
>
  <Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "marginBottom": 11.5,
          "marginLeft": 35,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "30%",
            },
          ]
        }
      />
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      />
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      />
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "backgroundColor": "#eee",
          "height": 1,
          "marginLeft": 35,
          "marginTop": 5,
          "width": "100%",
        }
      }
    />
  </Styled(ViewNB)>
</withAnimatable(Styled(ViewNB))>
`;

exports[`Testing BarIndicator Component renders with paragraph mode 1`] = `
<withAnimatable(Styled(ViewNB))
  animation="fadeIn"
  delay={0}
  direction="normal"
  duration={250}
  iterationCount={1}
  iterationDelay={0}
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onTransitionBegin={[Function]}
  onTransitionEnd={[Function]}
  useNativeDriver={false}
>
  <Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "marginHorizontal": 15,
          "marginTop": 15,
        }
      }
    >
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
            },
            Object {
              "width": "30%",
            },
          ]
        }
      />
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      />
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      />
      <Styled(ViewNB)>
        <Placeholder
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#eee",
                "borderRadius": 2.5,
                "height": 10,
                "justifyContent": "center",
                "marginTop": 10,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        />
        <Placeholder
          style={
            Array [
              Object {
                "alignSelf": "flex-start",
                "backgroundColor": "#eee",
                "borderRadius": 2.5,
                "height": 10,
                "justifyContent": "center",
                "marginTop": 10,
              },
              Object {
                "width": "100%",
              },
            ]
          }
        />
      </Styled(ViewNB)>
    </Styled(ViewNB)>
  </Styled(ViewNB)>
</withAnimatable(Styled(ViewNB))>
`;

exports[`Testing BarIndicator Component renders without animation 1`] = `
<Styled(ViewNB)>
  <Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "marginBottom": 11.5,
          "marginLeft": 35,
          "marginRight": 20,
          "marginTop": 5,
        }
      }
    >
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "30%",
            },
          ]
        }
      />
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      />
      <Placeholder
        style={
          Array [
            Object {
              "alignSelf": "flex-start",
              "backgroundColor": "#eee",
              "borderRadius": 2.5,
              "height": 10,
              "justifyContent": "center",
              "marginTop": 10,
            },
            Object {
              "width": "100%",
            },
          ]
        }
      />
    </Styled(ViewNB)>
    <Styled(ViewNB)
      style={
        Object {
          "backgroundColor": "#eee",
          "height": 1,
          "marginLeft": 35,
          "marginTop": 5,
          "width": "100%",
        }
      }
    />
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
